adept/_vlasov1d/solvers/pushers/fokker_planck.py

```diff
@@ -75,12 +80,109 @@ def __call__(
         """
 
         v0t_sq = self.vx_moment(f_xv * self.v[None, :] ** 2.0)
+
+        # Build coefficients without periodic boundaries
         a = nu[:, None] * dt * (-v0t_sq[:, None] / self.dv**2.0 + jnp.roll(self.v, 1)[None, :] / 2 / self.dv)
         b = 1.0 + nu[:, None] * dt * self.ones * (2.0 * v0t_sq[:, None] / self.dv**2.0)
         c = nu[:, None] * dt * (-v0t_sq[:, None] / self.dv**2.0 - jnp.roll(self.v, -1)[None, :] / 2 / self.dv)
+
         return a, b, c
 
 
+class ChangCooperLenardBernstein:
+    def __init__(self, cfg):
+        self.cfg = cfg
+        self.v = self.cfg["grid"]["v"]
+        self.dv = self.cfg["grid"]["dv"]
+        self.v_edge = 0.5 * (self.v[1:] + self.v[:-1])
+        self.ones = jnp.ones((self.cfg["grid"]["nx"], self.cfg["grid"]["nv"]))
+
+    def vx_moment(self, f_xv):
+        return jnp.sum(f_xv, axis=1) * self.dv
+
+    @staticmethod
+    def _chang_cooper_delta(w):
+        small = jnp.abs(w) < 1.0e-8
+        delta_small = 0.5 - w / 12.0 + w**3 / 720.0
+        delta_full = 1.0 / w - 1.0 / jnp.expm1(w)
+        return jnp.where(small, delta_small, delta_full)
+
+    def __call__(
+        self, nu: jnp.float64, f_xv: jnp.ndarray, dt: jnp.float64
+    ) -> tuple[jnp.ndarray, jnp.ndarray, jnp.ndarray]:
+        energy = self.vx_moment(f_xv * self.v[None, :] ** 2.0)
+        safe_energy = jnp.maximum(energy, 1.0e-30)
+
+        # Chang-Cooper weights are independent of nu because drift/diffusion share it.
+        w = -self.v_edge[None, :] * self.dv / safe_energy[:, None]
+        delta = self._chang_cooper_delta(w)
+
+        nu = nu[:, None]
+        diff = nu * energy[:, None]
+        drift = -nu * self.v_edge[None, :]
+        alpha = drift * delta + diff / self.dv
+        beta = drift * (1.0 - delta) - diff / self.dv
+
+        lam = dt / self.dv
+        a = jnp.zeros_like(self.ones)
+        a = a.at[:, 1:].set(-lam * alpha)
+        c = jnp.zeros_like(self.ones)
+        c = c.at[:, :-1].set(lam * beta)
+
+        beta_l = jnp.concatenate([jnp.zeros((self.cfg["grid"]["nx"], 1)), beta], axis=1)
+        alpha_r = jnp.concatenate([alpha, jnp.zeros((self.cfg["grid"]["nx"], 1))], axis=1)
+        diag = self.ones + lam * (alpha_r - beta_l)
+
+        return a, diag, c
+
+
+class ChangCooperDougherty:
```

Comment:
I am not familiar with the convention that

- Dougherty = `dv (T dv f + (v - u) f)`
whereas
- Lenard-Bernstein = `dv (T dv f + v f)` (i.e. u = 0).

I have always seen the nonzero mean velocity case described as Dougherty or L-B, interchangeably. I think you should make the zero mean velocity linearization explicit in the name of the class, or at the very least a comment.

Reply:
Speaking of which, there is an obvious code-sharing opportunity here -- why not have the `u=0` case pass a zero vbar/u array directly to this class?

Reply:
I'm still not convinced that DG/LB is widely understood as being the drift momentum split. Gkeyll calls it Lenard Bernstein everywhere, and their operator certainly includes drift velocity.